test/e2e/upgrades/topology_manager.go

```diff
@@ -0,0 +1,311 @@
+	return false
+}
+
+// Setup creates a pod requesting aligned resources.
+func (t *TopologyManagerUpgradeTest) Setup(f *framework.Framework) {
+	ginkgo.By("Creating a Pod with aligned resources")
+	node := getTopologyManagerSingleNumaNodePolicyEnabledNode(f)
+	if node == nil {
+		framework.Logf("No suitable node configured with Topology Manager before upgrade")
+		return
+	}
+
+	name := "pod-before-" + string(uuid.NewUUID())
+	err := runAndValidate(f, name, node)
+	framework.ExpectNoError(err)
+}
+
+// Test waits for the upgrade to complete, and then verifies that it is possible to run
+// a pod requesting aligned resources after the upgrade
+func (t *TopologyManagerUpgradeTest) Test(f *framework.Framework, done <-chan struct{}, upgrade UpgradeType) {
+	<-done
+	if !isRelevantUpgrade(upgrade) {
+		return
+	}
+
+	ginkgo.By("Verifying it is still possible to create a Pod with aligned resources")
+	node := getTopologyManagerSingleNumaNodePolicyEnabledNode(f)
+	if node == nil {
+		framework.Logf("No suitable node configured with Topology Manager before upgrade")
+		return
+	}
+
+	name := "pod-after-" + string(uuid.NewUUID())
+	err := runAndValidate(f, name, node)
+	framework.ExpectNoError(err)
+}
+
+// Teardown cleans up any remaining resources.
+func (t *TopologyManagerUpgradeTest) Teardown(f *framework.Framework) {
+	// rely on the namespace deletion to clean up everything
+}
+
+func runAndValidate(f *framework.Framework, name string, node *v1.Node) error {
+	pod := makePod(node.ObjectMeta.Name, f.Namespace.Name, name, testCmd)
+	pod = f.PodClient().CreateSync(pod)
+
+	output, err := e2epod.GetPodLogs(f.ClientSet, f.Namespace.Name, pod.Name, pod.Spec.Containers[0].Name)
+	framework.ExpectNoError(err, fmt.Sprintf("Failed to get pod %q output", name))
+
+	return validatePodOutput(output)
+}
+
+func validatePodOutput(output string) error {
+	cpuMap := make(map[string][]int)
+	re := regexp.MustCompile(`(\w*)=(\w*)`)
+	for _, match := range re.FindAllStringSubmatch(output, -1) {
+		if len(match) != 3 {
+			framework.Logf("unexpected match %v", match)
+			continue
+		}
+
+		cset, err := cpuset.Parse(match[2])
+		framework.ExpectNoError(err, "parsing %v", match)
+		// ToSlice() returns a sorted slice
+		cpuMap[match[1]] = cset.ToSlice()
+	}
+
+	allowed, ok := cpuMap["ALLOWED"]
+	if !ok {
+		return fmt.Errorf("no allowed CPUs found in output")
+	}
+
+	if len(allowed) < 2 {
+		return fmt.Errorf("the test requires at least two CPUs requested for the pod")
+	}
+
+	for i := 1; i < len(allowed); i++ {
+		cpuNumPrev := allowed[i-1]
+		sibilingsPrev, ok := cpuMap[fmt.Sprintf("SIBILING%d", cpuNumPrev)]
+		if !ok {
+			return fmt.Errorf("Unknown sibilings for cpu %d", cpuNumPrev)
+		}
+
+		cpuNum := allowed[i]
+		sibilings, ok := cpuMap[fmt.Sprintf("SIBILING%d", cpuNum)]
+		if !ok {
+			return fmt.Errorf("Unknown sibilings for cpu %d", cpuNum)
+		}
+
+		if !reflect.DeepEqual(sibilingsPrev, sibilings) {
```

Comment:
Does this require that all CPU cores should be in the same siblings from a pod viewpoint?
BTW typo: s/sibilings/siblings/

Reply:
fixed typo and clarified the intent of the test.